File: src/main/resources/archetype-resources/src/main/java/Connector.java
package ${groupId};

import java.util.Map;

import org.bonitasoft.engine.connector.AbstractConnector;
import org.bonitasoft.engine.connector.ConnectorException;
import org.bonitasoft.engine.connector.ConnectorValidationException;

public class Connector extends AbstractConnector {

    static final String DEFAULT_INPUT = "defaultInput";
    static final String DEFAULT_OUTPUT = "defaultOutput";

    @Override
    public void validateInputParameters() throws ConnectorValidationException {
        checkMandatoryStringInput(DEFAULT_INPUT);
    }

    protected void checkMandatoryStringInput(String inputName) throws ConnectorValidationException {
        try {
            String value = (String) getInputParameter(inputName);
            if (value == null || value.isEmpty()) {
                throw new ConnectorValidationException(this,
                        String.format("Mandatory parameter '%s' is missing.", inputName));
            }
        } catch (ClassCastException e) {
            throw new ConnectorValidationException(this, String.format("'%s' parameter must be a String", inputName));
        }
    }

    @Override
    protected void executeBusinessLogic() throws ConnectorException {
        System.out.println(String.format("Default input: %s", getInputParameter(DEFAULT_INPUT)));
        setOutputParameter(DEFAULT_OUTPUT, String.format("%s - output", getInputParameter(DEFAULT_INPUT)));
    }

}

File: src/main/resources/archetype-resources/src/main/kotlin/Connector.kt
package ${groupId}

import org.bonitasoft.engine.connector.AbstractConnector
import org.bonitasoft.engine.connector.ConnectorValidationException
import org.slf4j.LoggerFactory;

class Connector : AbstractConnector {

    constructor() : super()

    val logger = LoggerFactory.getLogger(Connector::class.java.name)
    val defaultInput = "defaultInput"

    override fun validateInputParameters() {
        checkMandatoryStringInput(defaultInput)
    }

    fun checkMandatoryStringInput(inputName: String) {
        val value = getInputParameter(inputName)
        if (!(value is String)) {
            throw ConnectorValidationException(this, "'$inputName' parameter must be a String.")
        } else if (value.isNullOrEmpty()) {
            throw ConnectorValidationException(this, "Mandatory parameter '$inputName' is missing.")
        }
    }

    override fun executeBusinessLogic() {
        val value = getInputParameter(defaultInput)
        logger.info("Default input: $value")
    }
}